root/process.py

```diff
@@ -0,0 +1,145 @@
+import os
+import zipfile
+import tempfile
+import hashlib
+import requests
+import ROOT
+import pandas as pd
+import logging
+from datetime import datetime
+import configparser
+from sqlalchemy import create_engine, MetaData, Table, Column, Integer, Float, String, DateTime, inspect
+
+log_file = 'processing.log'
+logging.basicConfig(filename=log_file, level=logging.INFO, format='%(asctime)s - %(message)s')
+
+def add_separator_to_log():
+    separator = "\n" + "=" * 80 + "\n" + f"New Run: {datetime.now()}" + "\n" + "=" * 80 + "\n"
+    with open(log_file, 'a') as log:
+        log.write(separator)
+
+def download_zip(url, download_path):
+    response = requests.get(url)
+    if response.status_code == 200:
+        with open(download_path, 'wb') as f:
+            f.write(response.content)
+        logging.info(f"Downloaded zip file from {url}")
+    else:
+        logging.error(f"Failed to download zip file from {url}, status code: {response.status_code}")
+        raise Exception(f"Failed to download zip file from {url}")
+    
+def read_db_config(filename='db_config.ini', section='postgresql'):
+    parser = configparser.ConfigParser()
+    parser.read(filename)
+
+    db_config = {}
+    if parser.has_section(section):
+        params = parser.items(section)
+        for param in params:
+            db_config[param[0]] = param[1]
+    else:
+        raise Exception(f'Section {section} not found in the {filename} file')
+
+    return db_config
+
+def hash_table_name(table_name, max_length=63):
+    if len(table_name) > max_length:
+        hash_object = hashlib.md5(table_name.encode())
+        hash_suffix = hash_object.hexdigest()[:8]
+        return f"{table_name[:max_length-9]}_{hash_suffix}"
+    return table_name
+
+def add_table_to_postgres(csv_file, table_name):
+    db_config = read_db_config()
+    db_uri = f"postgresql://{db_config['user']}:{db_config['password']}@{db_config['host']}:{db_config['port']}/{db_config['database']}"
+    engine = create_engine(db_uri)
+    connection = engine.connect()
+
+    metadata = MetaData()
+
+    df = pd.read_csv(csv_file)
+    
+    columns = []
+    for col_name, col_type in zip(df.columns, df.dtypes):
```

Comment:
postoji i nesto sto se zove infer schema, to moze dosta da ti olaksa ovaj deo sa types

Reply:
da, ocigledno df.to_sql samo napravi tabelu i prepozna tipove dovoljno dobro.. pa mi je taj deo skroz nepotreban